src/term/mod.rs

```diff
@@ -200,6 +200,65 @@ pub use self::mode::TermMode;
 
 pub const TAB_SPACES: usize = 8;
 
+#[derive(Clone, Copy)]
+enum Charset {
```

Comment:
I'd like to hold off on this particular addition until there's a way to use it. For the time being, it would just be added complexity.

Instead, let's use the `enum Charset` defined in _ansi.rs_. For mapping inputs, we could add a trait impl for it in this module.